### PR TITLE
#655 - Added toString() method to CollectionEnvelope class.

### PR DESCRIPTION
**Pull request by driver733**
As per #655 
I have added the `toString()` method to the parent class (`CollectionEnvelope`).
There is no need to add it to the `CollectionOf` class.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#697](https://github.com/yegor256/cactoos/issues/697) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#697](https://github.com/yegor256/cactoos/issues/697) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/512251274). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/java/org/cactoos/collection/CollectionEnvelope.java [hunk 1]
```diff
@@ -137,4 +137,11 @@ public final void clear() {
             "#clear(): the collection is ready only"
         );
     }
+
+    // @checkstyle DesignForExtensionCheck (5 lines)
```
Comment (amihaiemil) [inline src/main/java/org/cactoos/collection/CollectionEnvelope.java]: @driver733 Why is this needed here?

**Comment by amihaiemil** [comment]
@driver733 please add some unit tests, for CollectionOf  and CollectionEnvelope, to check that ``toString()`` is working for both of them. I would also expect edge cases, like empty collection :D

File: src/main/java/org/cactoos/collection/CollectionEnvelope.java (hunk 1 above)
Comment (driver733) [reply src/main/java/org/cactoos/collection/CollectionEnvelope.java]: @amihaiemil Because according to checkstyle,
The overridden methods should not have javadocs. 
And
>  If the method which can be overridden in a subclass has a javadoc comment (a good practice is to explain its self-use of overridable methods) the check will not rise a violation. The violation can also be skipped if the method which can be overridden in a subclass has one or more annotations that are specified in ignoredAnnotations option. Note, that by default @Override annotation is not included in the ignoredAnnotations set as in a subclass the method which has the annotation can also be overridden in its subclass.

**Comment by driver733** [comment]
@amihaiemil The `toString()` method does work for both of them.
The `CollectionOf` class does not incapsulate anything  by itself and uses the implementation of the parent class - `CollectionEnvelope`.
The CollectionEnvelope class uses the `toString ` method of `java.util.collection`, which does not need to be tested.

**Comment by amihaiemil** [comment]
@driver733 Yes, but I think a unit test is still needed to prove that

    a) it works for both of the classes (the fact that we can *see*, since the class is empty, is not an argument, by principle) and

    b)  the output's format is as the Issue's author asked for. Plus, it's not something that should change -- it's part of the API and users of cactoos may use ``toString()`` for logging, which in turn may be analyzed by parsers, in which case it is really not convenient if cactoos changes toString implementations.

**Comment by driver733** [comment]
@amihaiemil Done.

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by amihaiemil** [comment]
@yegor256 this also can be merged

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by yegor256** [comment]
@rultor merge

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code. Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review.
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka sure, I confirm :)

**Comment by ypshenychka** [comment]
@amihaiemil thanks!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
`QA` review completed: +8 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+718](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+1010](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/cactoos/issues/697#issuecomment-368482942))

@ypshenychka You can't do that, unless you have one of these roles: `ARC`, `PO`. Your current roles are: `QA`.

**Comment by 0crat** [comment]
This pull request [#697](https://github.com/yegor256/cactoos/issues/697) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol), here is [why](https://www.0crat.com/footprint/C63314D6Z/591814634). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by yegor256** [comment]
@0crat refuse

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/697#issuecomment-369356179))

@yegor256 The user @neonailol[/z](https://www.0crat.com/u/neonailol) resigned from [#697](https://github.com/yegor256/cactoos/issues/697), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/697#issuecomment-369356204))

@yegor256 The job [#697](https://github.com/yegor256/cactoos/issues/697) is now out of scope

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/697#issuecomment-369356204))

@yegor256 Job [#697](https://github.com/yegor256/cactoos/issues/697) was not in scope
